packages/analytics-browser/src/plugins/network-connectivity-checker.ts

```diff
@@ -0,0 +1,61 @@
+import { getGlobalScope } from '@amplitude/analytics-client-common';
+import { BeforePlugin, BrowserClient } from '@amplitude/analytics-types';
+import { BrowserConfig } from 'src/config';
+
+interface EventListener {
+  type: 'online' | 'offline';
+  handler: () => void;
+}
+
+export const networkConnectivityCheckerPlugin = (): BeforePlugin => {
+  const name = '@amplitude/plugin-network-checker-browser';
+  const type = 'before' as const;
+  const globalScope = getGlobalScope();
+  let eventListeners: EventListener[] = [];
+
+  const addNetworkListener = (type: 'online' | 'offline', handler: () => void) => {
+    if (globalScope) {
+      globalScope.addEventListener(type, handler);
+      eventListeners.push({
+        type,
+        handler,
+      });
+    }
+  };
+
+  const removeNetworkListeners = () => {
+    eventListeners.forEach(({ type, handler }) => {
+      if (globalScope) {
+        globalScope.removeEventListener(type, handler);
+      }
+    });
+    eventListeners = [];
+  };
+
+  const setup = async (config: BrowserConfig, amplitude: BrowserClient) => {
+    config.offline = !navigator.onLine;
+
+    addNetworkListener('online', () => {
+      config.offline = false;
+      // Flush immediately will cause ERR_NETWORK_CHANGED
+      setTimeout(() => {
+        amplitude.flush();
+      }, config.flushIntervalMillis);
```

Comment:
I feel this should be more immediate. Can we do next frame instead? `setTimeout(() => {...}, 1)`?

Reply:
Ok, I think that's ok. Can you please call this out in docs. Thanks

Reply:
Definitely!

Reply:
Docs PR: https://github.com/amplitude/amplitude-dev-center/pull/159